### PR TITLE
g++: add page

### LINKED ISSUE #1661: page request: g++
Webpage here - https://linux.die.net/man/1/g++

Please go through the [CONTRIBUTING](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) page to understand what is the format we use for our pages.

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
----

- [x] The page (if new), does not already exist in the repo.

- [x] The page (if new), has been added to the correct platform folder:  
      `common/` if it's common to all platforms, `linux/` if it's Linux-specific, and so on.

- [x] The page has 8 or fewer examples.

- [x] The PR is appropriately titled:  
      `<command name>: add page` for new pages, or `<command name>: <description of changes>` for pages being edited

- [x] The page follows the [contributing](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) guidelines 

Closes #1661

### CODE REVIEW

File: pages/common/g++.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# G Plus Plus
+
+> G++ is a compiler for C++. Member of GNU GCC.
```
Comment (agnivade) [inline pages/common/g++.md]: How does this sound -
```
> Compiles C++ source files.
> Part of GCC (GNU Compiler Collection).
```

File: pages/common/g++.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+- Compile a source code file into an executable binary:
+
+`g++ {{input_source.cpp}} -o {{output_executable}}`
```
Comment (agnivade) [inline pages/common/g++.md]: Change input_source.cpp => source.cpp everywhere.

File: pages/common/g++.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+- Activate output of (almost) all errors and warnings:
```
Comment (agnivade) [inline pages/common/g++.md]: Why "almost" ?

Probably `Display all warnings and errors in the output:` sounds better.

File: pages/common/g++.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+`g++ {{input_source.cpp}} -Wall -o {{output_executable}}`
+
+- Choose a language standard to compile for(C++98/C++11/C++14/C++17):
```
Comment (agnivade) [inline pages/common/g++.md]: Remove `(C++98/C++11/C++14/C++17)`

File: pages/common/g++.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+`g++ {{input_source.cpp}} -std={{language_standard}} -o {{output_executable}}`
```
Comment (agnivade) [inline pages/common/g++.md]: Change language_standard to `C++98|C++11|C++14|C++17`

File: pages/common/g++.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# G Plus Plus
```
[the remark this answers is not in the document]
Comment (nalinbhardwaj) [reply pages/common/g++.md]: The bot gives error "Title should be alphanumeric with dashes, underscores or spaces"

File: pages/common/g++.md (hunk 3 above)
Comment (nalinbhardwaj) [reply pages/common/g++.md]: `-Wextra` enables more warnings, and even with both it still doesn't cover [these warnings](https://gcc.gnu.org/onlinedocs/gcc-4.4.0/gcc/Warning-Options.html).

File: pages/common/g++.md (hunk 5 above)
Comment (nalinbhardwaj) [reply pages/common/g++.md]: There are many other C++ standards, it's hard to create an exhaustive list here.

File: pages/common/g++.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (agnivade) [reply pages/common/g++.md]: Ah dang it. The linter needs to be updated. Ok, lets change it to gplusplus then.

File: pages/common/g++.md (hunk 1 above)
Comment (nalinbhardwaj) [reply pages/common/g++.md]: Rest of the changes make sense, will make those soon.

File: pages/common/g++.md (hunk 5 above)
Comment (agnivade) [reply pages/common/g++.md]: Ah ok, ignore it then.

File: pages/common/g++.md (hunk 4 above)
Comment (agnivade) [reply pages/common/g++.md]: Let's keep this since there are other standards too.

File: pages/common/g++.md (hunk 3 above)
Comment (agnivade) [reply pages/common/g++.md]: Ok, we can keep the "almost" then.

**Comment by nalinbhardwaj** [comment]
I think this is good to merge then now @agnivade

File: pages/common/g++.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
+# Gplusplus
```
Comment (agnivade) [inline pages/common/g++.md]: Change it to => gplusplus

File: pages/common/g++.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+> Compiles C++ source files.
+> Part of GCC (GNU Compiler Collection).
+
+- Compile a source code file into an executable binary:
+
+`g++ {{source.cpp}} -o {{output_executable}}`
+
+- Activate output of (almost) all errors and warnings:
```
Comment (agnivade) [inline pages/common/g++.md]: Can we change it to `Display (almost) all warnings and errors in the output` ?

**Comment by agnivade** [comment]
Thank you for bearing with our review process. We require one more approval from a member and then this can be merged.

**Pull request merged**